src/lib.rs

```diff
@@ -109,7 +110,7 @@ where
 
     Defaults to "./trace-{unix epoch time}.json"
     */
-    pub fn file(mut self, file: String) -> Self {
+    pub fn file(mut self, file: PathBuf) -> Self {
```

Comment:
Maybe we should be consistent with the other file APIs and take an ` impl AsRef<Path>`, and turn it into a `PathBuf` for storage.

Something like
```rust
pub fn file<P: AsRef<Path>>(mut self, file: P) -> Self {
    self.out_file = Some(file.as_ref().into());
    self
}
```